stdlib/public/stubs/Random.cpp

```diff
@@ -42,6 +42,10 @@
 #include "swift/Runtime/Mutex.h"
 #include "../SwiftShims/Random.h"
 
+#if defined(__wasi__)
+#include <algorithm> // required for std::min
+#endif
```

Comment:
Just make this unconditional.  This is inside the implementation, so it is fine.